frontend/app/services/GuardianContentService.scala

```diff
@@ -26,37 +23,33 @@ class GuardianContentService(actorSystem: ActorSystem, executionContext: Executi
   implicit private val as = actorSystem
   override implicit val ec = executionContext
 
-  private def eventbrite: Future[Seq[Content]] = {
-    val enumerator = Enumerator.unfoldM(Option(1)) {
-      _.map { nextPage =>
-        for {
-          response <- eventbriteQuery(nextPage)
-        } yield {
-          val pagination = ContentAPIPagination(response.currentPage, response.pages)
-          Some(pagination.nextPageOpt, response.results)
-
+  private def eventbrite(nextPage: Int = 1, accumulatedEventbriteData: Seq[Content] = Nil): Future[Seq[Content]] = {
+    for {
+      response <- eventbriteQuery(nextPage)
+      rest <- {
+        val pagination = ContentAPIPagination(response.currentPage, response.pages)
+        val combined = response.results.toSeq.reverse ++ accumulatedEventbriteData // linked list - prepend the smaller amount
```

Comment:
would Vector be a better choice, to avoid all this reversing? E.g.

```
  private def eventbrite(nextPage: Int = 1): Future[Vector[Content]] = {
    for {
      response <- eventbriteQuery(nextPage)
      content <- {
        val pagination = ContentAPIPagination(response.currentPage, response.pages)
        val pageResults = response.results.toVector
        pagination.nextPageOpt match {
          case Some(nextPage) => eventbrite(nextPage).map(nextResults => pageResults ++ nextResults)
          case None => Future.successful(pageResults)
        }
      }
    } yield content
  }
```

Reply:
good point, it wasn't tail recursive anyway, due to the flatMap, and Vector simplifies the logic.  I have updated this and the one below according to your suggestion and it works ok locally.